### PR TITLE
Improve documentation and hide a helper function

### PR DESCRIPTION
**Pull request by kevinsung**
- Made some additions and changes to documentation.
- Removed the function `get_sparse_polynomial_tensor` from `__init__.py`, since it appears to be only a helper function. I also renamed it to `polynomial_tensor_sparse`. Actually, it seems to fit more in `utils/sparse_tools.py`, but it uses `get_fermion_sparse` which would cause a circular dependency.
- Removed some unnecessary code from `get_sparse_operator`.

### CODE REVIEW

**Comment by babbush** [comment]
Looks like Travis is no longer having problems.

File: src/openfermion/transforms/_verstraete_cirac.py [hunk 1]
```diff
@@ -26,8 +26,12 @@ def verstraete_cirac_2d_square(operator, x_dimension, y_dimension,
                                snake=False):
     """Apply the Verstraete-Cirac transform on a 2-d square lattice.
 
-    Currently only spinless models are supported. Furthermore, x_dimension
-    should be even.
+    Note that this transformation adds one auxiliary fermionic mode
+    for each mode already present, and hence it doubles the number of qubits
+    needed to represent the system.
+
+    Currently only supports even values of x_dimension and only works
```
Comment (idk3) [inline src/openfermion/transforms/_verstraete_cirac.py]: To be clear, odd y_dimension is also not supported, right?

File: src/openfermion/utils/_sparse_tools.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def jordan_wigner_ladder_sparse(n_qubits, tensor_factor, ladder_type):
         n_qubits(int): Number qubits in the system Hilbert space.
 
     Returns:
-        The corresponding SparseOperator.
+        The corresponding Scipy sparse matrix.
```
Comment (idk3) [inline src/openfermion/utils/_sparse_tools.py]: Very good to catch these, looks like the rest of us have missed it for a long time :P

File: src/openfermion/transforms/_verstraete_cirac.py (hunk 1 above)
Comment (kevinsung) [reply src/openfermion/transforms/_verstraete_cirac.py]: Actually, odd y_dimension is supported! It has to do with the way the auxiliary graph is constructed.

**Pull request merged**